llvm/lib/Target/AArch64/AArch64TargetTransformInfo.h

```diff
@@ -235,10 +235,17 @@ class AArch64TTIImpl : public BasicTTIImplBase<AArch64TTIImpl> {
   bool useNeonVector(const Type *Ty) const;
 
   InstructionCost
-  getMemoryOpCost(unsigned Opcode, Type *Src, MaybeAlign Alignment,
-                  unsigned AddressSpace, TTI::TargetCostKind CostKind,
-                  TTI::OperandValueInfo OpInfo = {TTI::OK_AnyValue, TTI::OP_None},
-                  const Instruction *I = nullptr);
+  getConstVectCost(unsigned Opcode, Type *Src, MaybeAlign Alignment,
+                   unsigned AddressSpace, TTI::TargetCostKind CostKind,
+                   TTI::OperandValueInfo OpInfo, const Instruction *I,
+                   const bool SrcIsConstVect, InstructionCost ScalarCost);
+
+  InstructionCost getMemoryOpCost(
+      unsigned Opcode, Type *Src, MaybeAlign Alignment, unsigned AddressSpace,
+      TTI::TargetCostKind CostKind,
+      TTI::OperandValueInfo OpInfo = {TTI::OK_AnyValue, TTI::OP_None},
+      const Instruction *I = nullptr, const bool SrcIsConstVect = false,
+      InstructionCost ScalarCost = 10000);
```

Comment:
I'm not sure we really want to be introducing magic values like this. We've been trying to remove/avoid this in the last few years now that InstructionCost can be marked as invalid. In fact, you can probably condense the two operands into one:

```
  InstructionCost getMemoryOpCost(
      unsigned Opcode, Type *Src, MaybeAlign Alignment, unsigned AddressSpace,
      TTI::TargetCostKind CostKind,
      TTI::OperandValueInfo OpInfo = {TTI::OK_AnyValue, TTI::OP_None},
      const Instruction *I = nullptr,
      InstructionCost ScalarCost = InstructionCost::getInvalid());
```

That way in the implementation of `getMemoryOpCost` you can ask if the cost is valid or not, which has the same effect as `SrcIsConstVect`.

Reply:
Passing `SrcIsConstVect` was intentional since otherwise we wouldn't know why the ScalarCost has been passed to the API.
About the magic no. Yes, will surely change it.

Reply:
Maybe I should change to `ConstVectorScalarCost` if thats what you are thinking?